plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/conversions/RedundantModifiersInOverrideConversion.kt

```diff
@@ -0,0 +1,138 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+
+package org.jetbrains.kotlin.nj2k.conversions
+
+import org.jetbrains.kotlin.analysis.api.KtAnalysisSession
+import org.jetbrains.kotlin.idea.base.psi.kotlinFqName
+import org.jetbrains.kotlin.lexer.KtModifierKeywordToken
+import org.jetbrains.kotlin.lexer.KtTokens.*
+import org.jetbrains.kotlin.nj2k.NewJ2kConverterContext
+import org.jetbrains.kotlin.nj2k.RecursiveConversion
+import org.jetbrains.kotlin.nj2k.symbols.JKClassSymbol
+import org.jetbrains.kotlin.nj2k.symbols.JKMultiverseClassSymbol
+import org.jetbrains.kotlin.nj2k.symbols.JKMultiverseKtClassSymbol
+import org.jetbrains.kotlin.nj2k.symbols.JKUniverseClassSymbol
+import org.jetbrains.kotlin.nj2k.tree.*
+import org.jetbrains.kotlin.nj2k.tree.OtherModifier.OVERRIDE
+import org.jetbrains.kotlin.nj2k.tree.Visibility.*
+import org.jetbrains.kotlin.nj2k.types.JKClassType
+import org.jetbrains.kotlin.nj2k.types.fqName
+import org.jetbrains.kotlin.nj2k.visibility
+import org.jetbrains.kotlin.psi.psiUtil.visibilityModifierTypeOrDefault
+
+
+class RedundantModifiersInOverrideConversion(context: NewJ2kConverterContext) : RecursiveConversion(context) {
```

Comment:
I wonder, can we simplify this by essentially reimplementing `org.jetbrains.kotlin.resolve.OverridingUtil#findMaxVisibility` over Java PSI (using `com.intellij.psi.PsiMethod#findSuperMethods()`). (`findMaxVisibility` is used in `org.jetbrains.kotlin.idea.core.PsiModificationUtilsKt#implicitVisibility`).

Reply:
I mean, `org.jetbrains.kotlin.idea.inspections.RedundantVisibilityModifierInspection.Holder#getRedundantVisibility` looks a lot simpler than the new conversion, and the hardest part is `OverridingUtil#findMaxVisibility`.